### PR TITLE
HLClient: First additions to search interface

### PR DESCRIPTION
**Pull request by cbuescher**
Implementing a few more of the access methods in ClientSearchHits/ClientSearchHit which are implementations of the common SearchHit(s) interfaces in core.

This is a PR against a feature branch.

### CODE REVIEW

File: client/highlevel-rest/src/test/java/org/elasticsearch/client/HLClientSearchIT.java [hunk 1]
```diff
@@ -39,34 +53,90 @@ public void init() {
...
+        mappingsBuilder.field("store", "true");
+        mappingsBuilder.endObject();
+        mappingsBuilder.startObject("content");
+        mappingsBuilder.field("type", "text");
+        mappingsBuilder.field("store", "true");
+        mappingsBuilder.endObject();
+        mappingsBuilder.endObject();
+        mappingsBuilder.endObject();
+        mappingsBuilder.endObject();
+        mappingsBuilder.endObject();
```
Comment (javanna) [inline client/highlevel-rest/src/test/java/org/elasticsearch/client/HLClientSearchIT.java]: if we'll need this in other tests, we should probably try to shorten this setup part of test by re-using what we have in our java api, that allows to provide `Object... source` , but we also want to be able to randomize the xcontent type, which is why we need to adapt it a bit

**Review by javanna — APPROVED** [review]
Left a few minors, along the same lines as the comments left on the initial commit of the feature branch. LGTM otherwise

File: client/highlevel-rest/src/main/java/org/elasticsearch/client/ClientSearchHit.java [hunk 2]
```diff
@@ -185,80 +189,97 @@ public Explanation explanation() {
...
     public String[] getMatchedQueries() {
-        // TODO
-        return null;
+        return matchedQueries();
     }
 
     @Override
     public SearchShardTarget shard() {
-        // TODO
+        // TODO, not sure this will work
```
Comment (cbuescher) [inline client/highlevel-rest/src/main/java/org/elasticsearch/client/ClientSearchHit.java]: Okay, should I remove this (and implementing the interface) in this PR or just change the TODO?

File: client/highlevel-rest/src/test/java/org/elasticsearch/client/HLClientSearchIT.java (hunk 1 above)
Comment (cbuescher) [inline client/highlevel-rest/src/test/java/org/elasticsearch/client/HLClientSearchIT.java]: Anything particular in mind? Or should I leave it for now and deal with it as soon as we need it more often?

Comment (javanna) [inline client/highlevel-rest/src/test/java/org/elasticsearch/client/HLClientSearchIT.java]: I can try and play with it afterwards, no big deal

**Pull request merged**